proteus/Gauges.py

```diff
@@ -25,67 +97,55 @@ class PointGauges(AV_base):
-                        activeTime=(0, 2.5),
-                        sampleRate=0.2,
-                        fileName='combined_gauge_0_0.5_sample_all.csv')
-
-        This creates a PointGauges object that will monitor the u and v fields at the locations [0.5, 0.5,
-        0] and [1, 0.5, 0], and the p field at [0.5, 0.5, 0] at simulation time between = 0 and 2.5 with samples
-        taken no more frequently than every 0.2 seconds.  Results will be saved to:
-        combined_gauge_0_0.5_sample_all.csv.
-
         Data is currently column-formatted, with 10 characters allotted to the time field, and 45 characters
         allotted to each point field.
         """
 
         AV_base.__init__(self)
-        self.gauges = gauges
-        self.measuredFields = set()
-
-        # build up dictionary of location information from gauges
-        # dictionary of dictionaries, outer dictionary is keyed by location (3-tuple)
-        # inner dictionaries contain monitored fields, and closest node
-        # closest_node is None if this process does not own the node
-        self.locations = OrderedDict()
-        for gauge in gauges:
-            fields, locations = gauge
-            self.measuredFields.update(fields)
-            for location in locations:
-                # initialize new dictionary of information at this location
-                if location not in self.locations:
-                    l_d = {'fields': set()}
-                    self.locations[location] = l_d
-                # add any currently unmonitored fields
-                self.locations[location]['fields'].update(fields)
 
+        self.fields = fields
         self.activeTime = activeTime
         self.sampleRate = sampleRate
         self.fileName = fileName
+        self.points = points if points else OrderedDict()
+        self.lines = lines if lines else []
         self.file = None  # only the root process should have a file open
         self.flags = {}
         self.files = {}
         self.outputWriterReady = False
         self.last_output = None
+        self.pointGaugeMats = []
+        self.field_ids = []
+        self.dofsVecs = []
+        self.pointGaugeVecs = []
+        self.segments = []
+
+        self.isPointGauge = bool(points)
+        self.isLineGauge = bool(lines)
+
+        if not (self.isPointGauge or self.isLineGauge):
+            raise ValueError("Need to provide points or lines")
+        if self.isPointGauge and self.isLineGauge:
+            raise ValueError("Must be one of point or line gauge but not both")
+
+    def getLocalNearestNode(self, location):
+        # determine local nearest node distance
+        node_distances = np.linalg.norm(self.vertices - location, axis=1)
+        nearest_node = np.argmin(node_distances)
+        nearest_node_distance = node_distances[nearest_node]
+        comm = Comm.get().comm.tompi4py()
+        return comm.rank, nearest_node, nearest_node_distance
 
     def findNearestNode(self, location):
```

Comment:
@cekees - I agree that this is something we want, but I'm going to defer this unless we need it, since it looks it's going to take me a little time to refactor the code in MeshTools if you need to preserve the functionality of the existing code.